src/FMData.Rest/DefaultAuthTokenProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FMData.Rest
+{
+    /// <summary>
+    /// Default authentication provider
+    /// </summary>
+    public class DefaultAuthTokenProvider : IAuthTokenProvider
+    {
+        private readonly ConnectionInfo _conn;
+
+        /// <summary>
+        /// Constructor 
+        /// </summary>
+        /// <param name="conn">Provide Connection details</param>
+        public DefaultAuthTokenProvider(ConnectionInfo conn)
+        {
+            _conn = conn;
+        }
+
+        /// <summary>
+        /// Connection config values
+        /// </summary>
+        public ConnectionInfo Conn { get => _conn; }
+
+        /// <summary>
+        /// Get base64 encoded AuthenticationHeaderValue
+        /// </summary>
+        /// <returns></returns>
+        /// <exception cref="System.NotImplementedException"></exception>
```

Comment:
Not needed, as this doesn't throw a NotImplementedException.

Reply:
I've changed it to ArgumentException. I don't know where that came from 🙈. But I kept the check - also because there're unit test's which check if an ArgumentException is thrown.